### PR TITLE
Added command line arguments for API test script

### LINKED ISSUE #102: Add command line arguments for API test script
In https://github.com/kruize/kruize/blob/master/tests/testAPI.sh
the URL, the port and the auth token are hardcoded in the script. Make changes so that command line arguments can be passed for the same

### PR DESCRIPTION
**Pull request by yathamravali**
Signed-off-by: Ravali Yatham <yatraval@in.ibm.com>

closes #102

### CODE REVIEW

File: tests/testAPI.sh [hunk 1]
```diff
@@ -21,6 +21,29 @@ KRUIZE_URL="http://localhost"
 KRUIZE_PORT="31313"
 AUTH_TOKEN=""
 
+function usage() {
+	echo "Usage: $0 [-u kruize_url] [-t kruize_port] [-a auth_token]"
```
Comment (ShishirH) [inline tests/testAPI.sh]: Kruize port should be -p

Comment (ShishirH) [reply tests/testAPI.sh]: Also please add `echo` above this line so that a new line might be printed

File: tests/testAPI.sh [hunk 2]
```diff
@@ -21,6 +21,29 @@ KRUIZE_URL="http://localhost"
...
+	exit -1
+}
+
+# Iterate through the commandline options
```
Comment (ShishirH) [inline tests/testAPI.sh]: This entire block should be moved below all the existing functions, ie, just above the line https://github.com/kruize/kruize/blob/2a27733eed20987b0e9249fecbf95907d4615689/tests/testAPI.sh#L71

File: tests/testAPI.sh [hunk 3]
```diff
@@ -21,6 +21,29 @@ KRUIZE_URL="http://localhost"
...
+while getopts u:t:a: gopts
```
Comment (ShishirH) [inline tests/testAPI.sh]: Nit: Can you please arrange this alphabetically, ie, a, t and u.

File: tests/testAPI.sh [hunk 4]
```diff
@@ -21,6 +21,29 @@ KRUIZE_URL="http://localhost"
...
+do
+	case ${gopts} in
+	u)
+		KRUIZE_URL="${OPTARG}"
+		;;
+	t)
```
Comment (ShishirH) [inline tests/testAPI.sh]: -p should be used for port

File: tests/testAPI.sh [hunk 5]
```diff
@@ -21,6 +21,29 @@ KRUIZE_URL="http://localhost"
...
+		KRUIZE_PORT="${OPTARG}"
+		;;
+        p)
```
Comment (ShishirH) [inline tests/testAPI.sh]: -a or -t should be used for the auth_token

File: tests/testAPI.sh (hunk 1 above)
Comment (yathamravali) [reply tests/testAPI.sh]: Done

File: tests/testAPI.sh (hunk 2 above)
Comment (yathamravali) [reply tests/testAPI.sh]: Moved curl_cmd as well to avoid further issues. Thanks!!

File: tests/testAPI.sh (hunk 3 above)
Comment (yathamravali) [reply tests/testAPI.sh]: Arranged alphabetically

File: tests/testAPI.sh [hunk 6]
```diff
@@ -21,6 +21,12 @@ KRUIZE_URL="http://localhost"
 KRUIZE_PORT="31313"
 AUTH_TOKEN=""
 
+function usage() {
+        echo
```
Comment (ShishirH) [inline tests/testAPI.sh]: Wrong indentation here (spaces instead of tab).

File: tests/testAPI.sh [hunk 7]
```diff
@@ -45,6 +49,26 @@ function get_list_of_applications {
 	${curl_cmd}/listApplications | jq -c '.[]' 
 }
 
+# Iterate through the commandline options
+while getopts a:p:u: gopts
+do
+        case ${gopts} in
```
Comment (ShishirH) [inline tests/testAPI.sh]: Spaces instead of tab in this entire loop

**Comment by ShishirH** [comment]
@yathamravali Please squash all the commits into a single one.
Pinging @dinogun for a final review and merge

**Pull request merged**